File: a-simple-util/src/main/java/io/github/xanderwang/asu/aLog.java
package io.github.xanderwang.asu;

import android.util.Log;

/**
 * log 工具类，用来打印 log
 *
 * @author xanderwang
 * @date: 20220108
 */
public class aLog {

    public static void d(String tag, String formatMsg, Object... args) {
        if (aConstants.logLevel <= Log.DEBUG) {
            Log.d(aUtil.logTag(tag), aUtil.format(formatMsg, args));
        }
    }

    public static void de(String tag, String errMsg, Throwable throwable) {
        if (aConstants.logLevel <= Log.DEBUG) {
            Log.d(aUtil.logTag(tag), errMsg, throwable);
        }
    }

    public static void w(String tag, String formatMsg, Object... args) {
        if (aConstants.logLevel <= Log.WARN) {
            Log.w(aUtil.logTag(tag), aUtil.format(formatMsg, args));
        }
    }

    public static void we(String tag, String errMsg, Throwable throwable) {
        if (aConstants.logLevel <= Log.WARN) {
            Log.w(aUtil.logTag(tag), errMsg, throwable);
        }
    }

    public static void e(String tag, String formatMsg, Object... args) {
        if (aConstants.logLevel <= Log.ERROR) {
            Log.e(aUtil.logTag(tag), aUtil.format(formatMsg, args));
        }
    }

    public static void ee(String tag, String errMsg, Throwable throwable) {
        if (aConstants.logLevel <= Log.ERROR) {
            Log.e(aUtil.logTag(tag), errMsg, throwable);
        }
    }
}

File: a-simple-util/src/main/java/io/github/xanderwang/asu/ALog.kt
package io.github.xanderwang.asu

import android.util.Log
import io.github.xanderwang.asu.AUtil.format
import io.github.xanderwang.asu.AUtil.logTag

/**
 * log 工具类，用来打印 log
 *
 * @author xanderwang
 * @date: 20220108
 */
object ALog {
    /**
     * log d
     *
     * @param tag
     * @param msg
     * @param args
     */
    @JvmStatic
    fun d(tag: String?, msg: String?, vararg args: Any?) {
        if (AConstants.logLevel <= Log.DEBUG) {
            Log.d(logTag(tag), format(msg, *args))
        }
    }

    /**
     * log d ，同时打印 error
     *
     * @param tag
     * @param msg
     * @param throwable
     */
    @JvmStatic
    fun d(tag: String?, msg: String?, throwable: Throwable) {
        if (AConstants.logLevel <= Log.DEBUG) {
            Log.d(logTag(tag), msg, throwable)
        }
    }

    /**
     * log w
     *
     * @param tag
     * @param msg
     * @param args
     */
    @JvmStatic
    fun w(tag: String?, msg: String?, vararg args: Any?) {
        if (AConstants.logLevel <= Log.WARN) {
            Log.w(logTag(tag), format(msg, *args))
        }
    }

    /**
     * log w
     *
     * @param tag
     * @param msg
     * @param throwable
     */
    @JvmStatic
    fun w(tag: String?, msg: String?, throwable: Throwable) {
        if (AConstants.logLevel <= Log.WARN) {
            Log.w(logTag(tag), msg, throwable)
        }
    }

    /**
     * log e
     *
     * @param tag
     * @param msg
     * @param args
     */
    @JvmStatic
    fun e(tag: String?, msg: String?, vararg args: Any?) {
        if (AConstants.logLevel <= Log.ERROR) {
            Log.e(logTag(tag), format(msg, *args))
        }
    }

    /**
     * log e
     *
     * @param tag
     * @param msg
     * @param throwable
     */
    @JvmStatic
    fun e(tag: String?, msg: String?, throwable: Throwable) {
        if (AConstants.logLevel <= Log.ERROR) {
            Log.e(logTag(tag), msg, throwable)
        }
    }
}